File: C++/2144. Minimum Cost of Buying Candies With Discount.cpp
class Solution {
public:
    int minimumCost(vector<int>& cost) {
        sort(cost.rbegin(), cost.rend());

        int res = 0, len = cost.size();

        for (int i = 0; i < len; i ++) {
            if (i % 3 == 2) continue;

            res += cost[i];
        }

        return res;
    }
};

File: Python/2144. Minimum Cost of Buying Candies With Discount.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        res = 0
        l = len(cost)

        cost.sort(reverse=True)

        for i in range(0, l, 3):
            res += cost[i]
            i += 1

            if i < l: res += cost[i]

        return res
